### PR TITLE
SelectWidget 

### PR DESCRIPTION
**Pull request by kmills006**
* If no default was provided to SelectWidget or default was an empty string, render empty option
* When computing defaults for bson form check the following conditions:
  * That the default is not an empty string
  * If the schema type is an enum
* If the schema type is an enum, return the default value as the type (int, double, long) or as the original value

### CODE REVIEW

**Comment by Chun-Yang** [comment]
Hi Kristy, can you talk to Lucas about your change? Just to make sure we do not introduce new edge cases. If you talk to Lucas about this and let him test on his cases for another related radar, we will merge this.

File: src/components/widgets/SelectWidget.js [hunk 1]
```diff
@@ -32,6 +36,11 @@ function SelectWidget({
   onChange
 }) {
   const {enumOptions} = options;
+
+  const emptyOption = !multiple && (!schema.default || schema.default.length === 0)
```
Comment (lablancas) [inline src/components/widgets/SelectWidget.js:40]: @kmills006 to confirm. This is meant to hide the empty option when a default is configured in the BSON schema or the default is configured as an empty string, correct? If that's the case, what if we show the empty option all the time unless it is a multiple select...?

**Review by lablancas — COMMENTED** [review]
See comments for SelectWidget.js (line 40); otherwise, looks good.

File: src/components/widgets/SelectWidget.js (hunk 1 above)
Comment (kmills006) [reply src/components/widgets/SelectWidget.js:40]: @lablancas That option should work to also. In this case, I followed the updates `react-jsonschema-form` has pushed since we forked which was to check if its a multiple select and if the default exists.
